tests/sp2n/Test_project_on_Sp.cc

```diff
@@ -55,35 +60,119 @@ int main (int argc, char **argv)
     
     std::cout << GridLogMessage << "Unitary matrix deformed " << std::endl;
     std::cout << GridLogMessage << "now U adjU - 1 = " << norm2(aux) << std::endl;
+    
     std::cout <<GridLogMessage << std::endl;
-
+    std::cout << GridLogMessage << "#   #   #   #" << std::endl;
+    std::cout << GridLogMessage << "Group" << std::endl;
+    std::cout << GridLogMessage << "#   #   #   #" << std::endl;
     std::cout <<GridLogMessage << std::endl;
-    std::cout << GridLogMessage << "Projecting on Sp2n " << std::endl;
-
+    //  Testing ProjectOnSpGroup
```

Comment:
All the following is a big duplication. We should attempt to make this more concise and less error-prone. My suggestion is something along the lines of:
```c++
const std::vector<std::tuple<std::string, std::function(LatticeColourMatrixD(LatticeColourMatrixD)>> name_and_function = {
    {"ProjectOnSpGroup", ProjectOnSpGroup},
    {"ProjectOnGaugeGroup", [](auto U) {ProjectOnGaugeGroup(U); return U;}},
    ...
};
for (auto& [name, projection] : name_and_function) {
    ... (run projection stuff) ...
}
```
A few things to note:
* There are other constructs in order to store functions. `std::function` is a simple version.
* I explicitly want no reference in the function argument and a return value. That way we can reuse the original `U` for all tests.
* The above version can easily be wrapped in another loop having different values for `U` (e.g. negative determinant).
* If there are peculiar things (like the additional test about the determinant), we can run them separately after the loop.
* The nice "Structured binding declaration `[name, projection] = ...` is C++17, I believe. Maybe it will get a tiny bit uglier than that.

Reply:
I've restructured the test a bit. nothing fancy but it should be tidier now